distributed/hdfs.py

```diff
@@ -283,3 +284,58 @@ def write_bytes(path, futures, executor=None, hdfs=None, **hdfs_auth):
         filenames = [os.path.join(path, template % i) for i in range(n)]
 
     return executor.map(write_block_to_hdfs, filenames, futures, hdfs=hdfs)
+
+
+@gen.coroutine
+def _read_text(fn, encoding='utf-8', errors='strict', lineterminator='\n',
+               executor=None, hdfs=None, lazy=False, collection=True):
+    from hdfs3 import HDFileSystem
+    from dask import do
+    import pandas as pd
+    hdfs = hdfs or HDFileSystem()
+    executor = default_executor(executor)
+
+    filenames = sorted(hdfs.glob(fn))
+    blocks = [block for fn in filenames
+                    for block in read_bytes(fn, executor, hdfs, lazy=True,
+                                            delimiter=lineterminator)]
```

Comment:
Possible problem: lineterminator may occur within a multi-byte group, i.e., it would not be a line terminator after decode. I don't know if this is allowed, or if unicode byte groups are designed to avoid it.